kf_lib_data_ingest/etl/ingest_pipeline.py

```diff
@@ -171,41 +167,16 @@ def run(self):
             # Iterate over stages and execute them
             output = None
             for stage in self._iterate_stages():
+                self.stages[stage.stage_type] = stage
                 if not output:  # First stage gets no input
                     output, concept_discovery_dict = stage.run()
                 else:
                     output, concept_discovery_dict = stage.run(output)
 
-                # Collapse stage subtypes to their bases for storage
-                # (e.g. GuidedTransformStage -> TransformStage)
-                stage_type = type(stage)
-                while stage_type.__base__ != IngestStage:
-                    stage_type = stage_type.__base__
-
-                self.stage_outputs[stage_type] = output
-
+                # Standard stage output validation
                 if concept_discovery_dict:
```

Comment:
Maybe don't bother to return it if the return isn't going to be used. check stage.concept_discovery_dict instead?